### ISSUE TITLE
Upgrade to Terraform 0.9.6

### ISSUE THREAD

**Issue by alexsomesan** [coreos/tectonic-installer#687]
Creating this now for visibility.

Terraform 0.9.5 has just been released a few days ago. Unfortunately, it's plagued by a couple of regressions, [one of which](https://github.com/hashicorp/terraform/issues/14399) is causing errors when `plan`ning the tectonic templates:
```
Error asking for user input: 2 error(s) occurred:

* module.vpc.var.master_azs: At column 15, line 1: true and false expression types must match; have type string and type unknown in:

${ split("|", "${length(keys(var.tectonic_aws_master_custom_subnets))}" > 0 ?
    join("|", keys(var.tectonic_aws_master_custom_subnets)) :
    join("|", data.aws_availability_zones.azs.names)
  )}
* module.vpc.var.worker_azs: At column 15, line 1: true and false expression types must match; have type string and type unknown in:

${ split("|", "${length(keys(var.tectonic_aws_worker_custom_subnets))}" > 0 ?
    join("|", keys(var.tectonic_aws_worker_custom_subnets)) :
    join("|", data.aws_availability_zones.azs.names)
  )}
```

I suggest to skip upgrading to 0.9.5 and wait 0.9.6, since tip of master is already able to build tectonic correctly.

**Comment by tothandras** [comment]
We are having the same problem within our own terraform script, a workaround is to force the type change in the ternary. Wrapping one of the clause in `format("%s", wrapThis)` fixes the issue.

**Comment by alexsomesan** [comment]
@tothandras Andras, thanks for sharing the workaround. It's great to have the option at hand for users that are blocked by this issue. The root cause of the problem seems to be a regression in terraform, which they already fixed and will be included in 0.9.6. 

Our plan is to sit 0.9.5 out and wait for 0.9.6 which will also include one contribution we made that will relieve us from shipping custom plugins after 0.9.6 comes out.
